lib/wegift/models/remote_code.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'faraday'
+require 'uri'
+
+module Wegift
+  class RemoteCode < Response
+    attr_accessor :url
+
+    # response/success
+    attr_accessor :amount, :barcode_format, :barcode_string, :code, :expiry_date,
+      :pin, :type
+
+    def self.get(url)
+      uri = URI(url)
+      conn = Faraday.new(url: "#{uri.scheme}://#{uri.host}")
+
+      response =  conn.get("#{uri.path}?format=json") do |req|
+        req.headers['Content-Type'] = 'application/json'
+      end
+
+      remote_code = self.new(url: url)
+      remote_code = remote_code.parse(response)
```

Comment:
why not just `remote_code.parse(response)`

Reply:
yeah I don't even know why I did that :/